src/plugins/orangehrmBuzzPlugin/Dto/BuzzPostShareSearchFilterParams.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA
+ */
+
+namespace OrangeHRM\Buzz\Dto;
+
+use OrangeHRM\Core\Dto\FilterParams;
+
+class BuzzPostShareSearchFilterParams extends FilterParams
+{
+    public const ALLOWED_SORT_FIELDS = [];
+
+    protected ?int $postId = null;
```

Comment:
don't make this nullable.
```suggestion
    protected int $postId;
```
refactor setters and getters.

e.g.
https://github.com/orangehrm/orangehrm/blob/5.3/src/plugins/orangehrmBuzzPlugin/Dto/BuzzFeedFilterParams.php#L29

Reason: this should be required parameter according to DAO code. so if other developer going to reuse this method they should set required parameters. otherwise compiler will throw non initialize error

Reply:
ah yes true. I'll add the changes in the next PR